test/unit/org/apache/cassandra/index/sai/cql/StorageAttachedIndexDDLTest.java

```diff
@@ -278,43 +326,757 @@ public void shouldFailCreationOnMultipleColumns()
+        }
+        flush();
+
+        Injections.Barrier delayIndexBuilderCompletion = Injections.newBarrier("delayIndexBuilder", 2, false)
+                                                                   .add(InvokePointBuilder.newInvokePoint().onClass(StorageAttachedIndexBuilder.class).onMethod("build"))
+                                                                   .build();
+
+        Injections.inject(delayIndexBuilderCompletion);
+        String indexName = createIndex(String.format(CREATE_INDEX_TEMPLATE, "v1"));
+        waitForAssert(() -> assertEquals(1, delayIndexBuilderCompletion.getCount()));
+
+        dropIndex("DROP INDEX %s." + indexName);
+
+        // let blocked builders to continue
+        delayIndexBuilderCompletion.countDown();
+        waitForCompactions();
+
+        delayIndexBuilderCompletion.disable();
+
+        verifySSTableIndexes(indexName, 0);
+        assertFalse("Expect index not built", SystemKeyspace.isIndexBuilt(KEYSPACE, indexName));
+
+        // create index again, it should succeed
+        indexName = createIndex(String.format(CREATE_INDEX_TEMPLATE, "v1"));
+        waitForIndexQueryable();
+        verifySSTableIndexes(indexName, 1);
+
+        ResultSet rows = executeNet("SELECT id1 FROM %s WHERE v1='0'");
+        assertEquals(num, rows.all().size());
+    }
+
+    @Test
+    public void nodetoolStopInitialIndexBuild() throws Throwable
+    {
+        createTable(CREATE_TABLE_TEMPLATE);
+        disableCompaction(KEYSPACE);
+
+        // create 100 rows into 1 sstable
+        int num = 100;
+        int sstable = 1;
+        for (int i = 0; i < num; i++)
+        {
+            execute("INSERT INTO %s (id1, v1) VALUES ('" + i + "', '0');");
+        }
+        flush();
+
+        Injections.Barrier delayIndexBuilderCompletion = Injections.newBarrierAwait("delayIndexBuilder", 1, true)
+                                                                   .add(InvokePointBuilder.newInvokePoint().onClass(StorageAttachedIndexBuilder.class).onMethod("build"))
+                                                                   .build();
+
+        Injections.inject(delayIndexBuilderCompletion);
+
+        IndexContext literalIndexContext = createIndexContext(createIndex(String.format(CREATE_INDEX_TEMPLATE, "v1")), UTF8Type.instance);
+
+        waitForAssert(() -> assertTrue(getCompactionTasks() > 0), 1000, TimeUnit.MILLISECONDS);
+
+        // Stop initial index build by interrupting active and pending compactions
+        int attempt = 20;
+        while (getCompactionTasks() > 0 && attempt > 0)
+        {
+            // only interrupts active compactions, not pending compactions.
+            CompactionManager.instance.stopCompaction(OperationType.INDEX_BUILD.name());
+            // let blocked builder to continue, but still block pending builder threads
+            delayIndexBuilderCompletion.reset();
+
+            Thread.sleep(3000);
```

Comment:
nit: Throwing `@SuppressWarnings("BusyWait")` on the method would stop tooling complaining about this.